libraries/aws_db_subnet_group.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsDbSubnetGroup < AwsResourceBase
+  name 'aws_db_subnet_group'
+  desc 'Verifies settings for a Subnet Group'
+  example "
+    describe aws_subnet_group(db_subnet_group_name: 'subnet-group') do
+      it { should exist }
+    end
+  "
+
+  def initialize(opts = {})
+    opts = { db_subnet_group_name: opts } if opts.is_a?(String)
+    super(opts)
+    validate_parameters(required: [:db_subnet_group_name])
+    @display_name = opts[:db_subnet_group_name]
+
+    catch_aws_errors do
+      resp = @aws.rds_client.describe_db_subnet_groups({ db_subnet_group_name: opts[:db_subnet_group_name] })
+      @subnet_group = resp.db_subnet_groups[0].to_h
+
+      create_resource_methods(@subnet_group)
+    end
+  end
+
+  def exists?
+    !@subnet_group.nil? && !@subnet_group.empty?
+  end
+
```

Comment:
I think it would be useful to overwrite the `subnets` method created by dynamics to return:
`[{:subnet_identifier=>"subnet-e2d4ff12", :subnet_availability_zone=>{:name=>"us-east-2c"}, :subnet_status=>"Active"}]` 
instead of 
`[#<#<Class:0x00007fc93b494b50>::AwsResourceProbe:0x00007fc91bbb8f78 @count=3, @item={:subnet_identifier=>"subnet-e2d4ffaf", :subnet_availability_zone=>{:name=>"us-east-2c"}, :subnet_status=>"Active"}`

This will allow using the `aws_subnet` resource more intuitively:
```ruby
aws_db_subnet_group('my-db-group').subnets.each do |subnet|
  describe aws_subnet(subnet[:subnet_identifier]) do
    its('cidr_block') { should eq '10.0.1.0/24' }
  end
end
```